fraqbot/Local/newjob.py

```diff
@@ -41,27 +42,53 @@ def listening_for(self, message):
                 (text.startswith('!job') or
                 text.startswith('!newjob')))
 
+    def _strip_and_parse_terms(self, term):
+        term_split = [
+            t.lower().strip() for t in
+            term.split(',') if len(t.strip()) > 0
+        ]
+        if len(term_split) > 2:
+            term_split[2] = re.sub(
+                '^at',
+                '',
+                term_split[2]).strip()
+        return [
+            # Role Modifier search term
+            term_split[0],
+            # Role search term
+            term_split[1] if
+            (len(term_split) > 1)
+            else term_split[0],
+            # Company search term
+            term_split[2] if
+            (len(term_split) > 2)
+            else term_split[0],
+        ]
+
     def _get_job(self, term):
 
         found_role_modifiers = []
         found_roles = []
         found_companies = []
 
         if term:
-            term_lowercase = term.lower()
+            search_terms_parsed = self._strip_and_parse_terms(term)
             found_role_modifiers = [
                     phrase for phrase in
-                    self.role_modifiers_list if term_lowercase in
+                    self.role_modifiers_list if
+                    search_terms_parsed[0] in
                     phrase.lower()
                 ]
             found_roles = [
                     phrase for phrase in
-                    self.roles_list if term_lowercase in
+                    self.roles_list if
+                    search_terms_parsed[1] in
                     phrase.lower()
                 ]
             found_companies = [
                     phrase for phrase in
-                    self.companies_list if term_lowercase in
+                    self.companies_list if
+                    search_terms_parsed[2] in
                     phrase.lower()
                 ]
```

Comment:
Since we're using the same logic over and over here, we should pop this out into its own method.
- Pass in a term and a list (or name of a class property that contains a list if you want to get fancy)
- Get back a new list